test/e2e/scrapeconfig_test.go

```diff
@@ -462,8 +462,8 @@ func testScrapeConfigDNSSDConfig(t *testing.T) {
 	sc.Spec.DNSSDConfigs = []monitoringv1alpha1.DNSSDConfig{
 		{
 			Names: []string{"node.demo.do.prometheus.io"},
-			Type:  ptr.To("A"),
-			Port:  ptr.To(9100),
+			Type:  ptr.To(monitoringv1alpha1.DNSSDType("A")),
+			Port:  ptr.To(int32(9100)),
 		},
```

Comment:
can we include a test with invalid type to make sure it doesn't accept that?

Reply:
Sure, I will add a test for invalid record types.

Reply:
Since we add the `Type` variable as a string of type `DNSRecordTypes`, I think we are not getting any error. Should we maybe also add enums? I tried doing this manually:
```
apiVersion: monitoring.coreos.com/v1alpha1
kind: ScrapeConfig
metadata:
  name: dns-sd
  namespace: default
  labels:
    prometheus: system-monitoring-prometheus
    app.kubernetes.io/name: scrape-config-example
spec:
  dnsSDConfigs:
    - names: ['http://my-external-api/discovery']
      type: WRONG
```
And it gets accepted.

Reply:
Thats why I thought we should have e2e test. Why we need constants?
We can do something similar to Role (only uppercase)?
https://github.com/prometheus-operator/prometheus-operator/blob/d285820ce6d2803c9c43ad7bdcbac9b11bf6311b/pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go#L76-L77

Reply:
I added the Enum validation and as expected, this same yaml manifest is rejected. Hmm... Am I missing something or does creating a type does not work as we intended it to work. If the latter is the case, we will have to touch upon other SDs too. (eg. ScaleWay, where we did something similar)

Reply:
You meant removed the constants and kept just enums? Did you run operator with updated code?
Enums should work.

Reply:
> I added the Enum validation and as expected, this same yaml manifest is rejected.

Yes, with the Enums it works as expected.

Reply:
Ok, we can keep the constants for validation. For validation better to have enum itself